[Core] Fix NuGet packaging project not creating .nupkg

mrward opened the pull request:
The packaging project type indicated that it only worked with
MSBuild through the project model MSBuild item types:

<Extension path="/MonoDevelop/ProjectModel/MSBuildItemTypes">
	<DotNetProjectType
		language="NuGet.Packaging"
		extension="nuproj"
		guid="{5DD5E4FA-CB73-4610-85AB-557B54E96AA9}"
		type="MonoDevelop.Packaging.PackagingProject"
		alias="NuGetPackaging"
		msbuildSupport="Required" />
</Extension>

Whilst the msbuildSupport property is obsolete the logic seems
to have changed and broken this so the NuGet packaging project
was not using MSBuild so nothing was being compiled. The check
to see if MSBuild should be used now handles the MSBuild support
being set to Required.

Fixes VSTS #650951 - Building NuGet packaging project does not
generate .nupkg

Also enabled some more packaging tests that should detect this problem happening again. Needs a new VSTS test run step - there is a clone of the VS Mac VSTS build with a new NuGetizer step which I am trying out.

mrward committed:
[Packaging] Enable more tests

Some tests were not being run when the NUnit exclude was used even
though the platform should not have been excluded. Instead the tests
now use MonoDevelop's Platform class to detect the platform and then
Assert.Ignore if the platform is not supported. Also enabled the
build test BuildPackagingProjectFromTemplate which is currently failing
since no .nupkg file is generated.

mrward committed:
[Core] Fix NuGet packaging project not creating .nupkg

The packaging project type indicated that it only worked with
MSBuild through the project model MSBuild item types:

<Extension path="/MonoDevelop/ProjectModel/MSBuildItemTypes">
	<DotNetProjectType
		language="NuGet.Packaging"
		extension="nuproj"
		guid="{5DD5E4FA-CB73-4610-85AB-557B54E96AA9}"
		type="MonoDevelop.Packaging.PackagingProject"
		alias="NuGetPackaging"
		msbuildSupport="Required" />
</Extension>

Whilst the msbuildSupport property is obsolete the logic seems
to have changed and broken this so the NuGet packaging project
was not using MSBuild so nothing was being compiled. The check
to see if MSBuild should be used now handles the MSBuild support
being set to Required.

Fixes VSTS #650951 - Building NuGet packaging project does not
generate .nupkg

mrward committed:
[Packaging] Fix NuGet package project template build test

The test was not setting the IncludeInPackage MSBuild metadata
for the readme.txt file so the build would fail since no items
were included in the NuGet package (.nupkg). Now the test uses
the project template wizard to set this property.

Therzok commented on main/src/addins/MonoDevelop.Packaging/MonoDevelop.Packaging.Tests/MonoDevelop.Packaging.Tests/AddPlatformImplementationTests.cs:
```diff
@@ -38,9 +39,18 @@ namespace MonoDevelop.Packaging.Tests
 		protected override void InternalSetup (string rootDir)
 		{
 			base.InternalSetup (rootDir);
+			Xwt.Application.Initialize (Xwt.ToolkitType.Gtk);
```
Remove these two and make it an `IdeTestBase`.

Therzok commented on main/src/addins/MonoDevelop.Packaging/MonoDevelop.Packaging.Tests/MonoDevelop.Packaging.Tests.csproj:
```diff
@@ -62,6 +63,10 @@
     <ProjectReference Include="..\..\MonoDevelop.DotNetCore\MonoDevelop.DotNetCore.csproj">
       <Project>{6868153E-41EA-43A4-A81A-C1E7256373F7}</Project>
       <Name>MonoDevelop.DotNetCore</Name>
```
This should be Private false

mrward replied:
The tests do not work when it is private false which is why I changed it.

Therzok replied:
Ah, I see why, so, the packaging tests use the dotnetcore addin, but the packaging addin does not depend on the dotnetcore addin.

mrward replied:
Yeah, just going to see what happens if I remove that code that directly uses the dotnetcore addin.

mrward replied:
Mmm, seems to work now. Something must have changed :)

Therzok commented:
Comments posted, once those are fixed, :shipit:

mrward committed:
[Packaging] Use IdeTestsBase in tests.

mrward committed:
[Packaging] Remove use of .NET Core addin from tests

The tests seem to work without this and the test project can now
avoid having a local copy of the .NET Core addin assembly.

Therzok commented on main/src/addins/MonoDevelop.Packaging/MonoDevelop.Packaging.Tests/MonoDevelop.Packaging.Tests/AddPlatformImplementationTests.cs:
```diff
@@ -36,21 +38,16 @@
-			#pragma warning disable 219
-			// Ensure MSBuildSdksPath is registered otherwise the project builders are recycled
-			// when we try to build the packaging project which breaks the tests.
-			string directory = DotNetCore.DotNetCoreSdk.MSBuildSDKsPath;
```
Should this have been removed?

mrward replied:
Yeah - not needed anymore it seems.

Therzok replied:
Nevermind, looks good!

Therzok commented on main/src/addins/MonoDevelop.Packaging/MonoDevelop.Packaging.Tests/MonoDevelop.Packaging.Tests/ProjectTemplateTests.cs:
```diff
@@ -38,19 +40,8 @@
-			#pragma warning disable 219
-			// Ensure MSBuildSdksPath is registered otherwise the project builders are recycled
-			// when we try to build the packaging project which breaks the tests.
-			string directory = DotNetCore.DotNetCoreSdk.MSBuildSDKsPath;
```
Should this have been removed?

Therzok replied:
Nevermind, looks good!

mrward commented:
@monojenkins backport release-7.6
